RightHandAngleHintHider

DorianRec opened the pull request:
/**
 * A type of {@link Hider}, generating a valid {@link AngleHint},
 * which contains an angle of the provided {@link PreferenceService#ANGLE} (standard is {@link Math#PI})
 * with his right wing pointing at the treasure.
*/

Commit:
implemented

ghost commented:
<img src="https://www.deepcode.ai/icons/green_check.svg" width= "50px" align= "left"/> Congratulations :tada:. DeepCode [analyzed](https://www.deepcode.ai/app/gh/jotoh98/treasureHunt/36b0991b2b930e29d29fc8f0b42a712007f71f60/jotoh98/treasureHunt/66054ca845c047fe45303e7802be378d8d3c9712/_/%2F/code/?utm_source=gh_review) your code in 0.366 seconds and we found no issues. Enjoy a moment of no bugs :sunny:.

##### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/jotoh98/treasureHunt/36b0991b2b930e29d29fc8f0b42a712007f71f60/jotoh98/treasureHunt/66054ca845c047fe45303e7802be378d8d3c9712/_/%2F/code/?utm_source=gh_review)
---

Commit:
Merge branch 'master' into brunch/RightHandAngleHintHider

1-alex98 commented:
@DorianRec resolve conflicts

DorianRec committed:
Merge branch 'master' into brunch/RightHandAngleHintHider

Commit:
Merge branch 'master' into brunch/RightHandAngleHintHider

Commit:
Merge branch 'master' into brunch/RightHandAngleHintHider

Commit:
Merge branch 'master' into brunch/RightHandAngleHintHider

Commit:
Merge branch 'master' into brunch/RightHandAngleHintHider

rukemn committed:
changed Angle preference from Radians to Degree
Merged Preference "Angle" and "preferred angle size"
if angle == 180, explicit HalfPlane is returned

rukemn committed:
deleted the Angle reference

rukemn committed:
docs fix

Commit:
javadoc fixed

Commit:
Merge branch 'master' into brunch/RightHandAngleHintHider
